Go survey: 1.21.0

Cyanoxygen opened the pull request:
Topic Description
-----------------

This topic updates the Go toolchain and Go packages we provide to their latest version.

Package(s) Affected
-------------------

### Existing packages

- `go` 1.21.0
- `podman` 4.6.2
- `android-platform-tools` 34.0.4
- `git-lfs` 3.4.0
- `delve` 1.21.0
- `libguestfs` 1.50.1
- `clash` 1.18.0
- `dnscrypt` 2.1.5
- `frp` 0.51.3
- `geoipupdate` 6.0.0
- `kcptun` 20230811
- `syncthing` 1.24.0
- `v2ray` 5.7.0+git20230810
- `v2ray-plugin` 1.3.2
- `xray` 1.8.4
- `direnv` 2.32.3
- `fzf` 0.42.0
- `pcstat` 0.0.2
- `restic` 0.16.0
- `hub` 2.14.2+git20230617
- `hugo` 0.118.2
- `rclone` 1.63.1
- `poetry-code` 1.7.0
- `poetry` 1.6.1
- `setuptools-scm` 7.1.0
- `cython` 3.0.0
- `cleo` 2.0.1
- `mongodb` 7.0.1
- `docker` 24.0.6
- `conmon` 2.1.8
- `criu` 3.18

### New

- `hatch-vcs` 0.3.0
- `scikit-build` 0.17.6
- `rapidfuzz-cpp` 2.0.0
- `rapidfuzz` 3.2.0
- `dulwich` 0.21.6
- `smmap` 5.0.0
- `gitdb` 4.0.10
- `gitpython` 3.1.35
- `mongo-tooling-metrics` 1.0.8
- `docker-compose` 2.1.0

Security Update?
----------------

No

Build Order
-----------

```
go conmon podman android-platform-tools git-lfs delve libguestfs clash dnscrypt frp geoipupdate kcptun syncthing v2ray v2ray-plugin xray direnv fzf pcstat restic hub hugo rclone git-lfs dnscrypt libguestfs poetry-core poetry setuptools-scm hatch-vcs scikit-build cython rapidfuzz-cpp rapidfuzz cleo dulwich smmap gitdb gitpython mongo-tooling-metrics mongodb criu docker docker-compose
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
- [x] Architecture-independent `noarch`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
- [ ] 32-bit Optional Environment `optenv32`
- [ ] Architecture-independent `noarch`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Cyanoxygen committed:
golang: update to 1.21.0

- Split $VER to three parts.
- Update Golang tools to 0.12.0.
- Update net library to 0.14.0.

Cyanoxygen committed:
go: update list of extra Go tools

- cover is now in pkg/tool.

Cyanoxygen committed:
go: replace yes and no with 1 and 0 in defines

Cyanoxygen committed:
go: install binaries to /usr/lib/go/bin/

- Before this, go and gofmt was installed to /usr/bin, which confuses Go
  as it can not figure out $GOROOT automatically by os.Executable, see
  [1] for details.
  By installing binaries to $GOROOT/bin (GOROOT points to /usr/lib/go),
  go can now figure out its GOROOT.

- Provide symlinks in /usr/bin/go{,fmt}-google to ensure it is there.

- Update alternatives file to point corresponding executables to the
  one installed in $GOROOT.

[1]: https://github.com/golang/go/issues/61921#issuecomment-1673399370

Cyanoxygen committed:
go: install go.env file

Cyanoxygen committed:
go: unlink /usr/lib/go/bin in preinst

- So that go binaries does not install on /usr/bin. dpkg does not unlink
  this, and creats hazards for alternatives.

Cyanoxygen committed:
podman: update to 4.6.2

Cyanoxygen committed:
android-platform-tools: update to 34.0.4

Cyanoxygen committed:
delve: update to 1.21.0

Cyanoxygen committed:
git-lfs: update to 3.4.0

Cyanoxygen committed:
v2ray-plugin: update to 1.3.2

Cyanoxygen committed:
geoipupdate: update to 6.0.0

Cyanoxygen committed:
kcptun: update to 20230811

Cyanoxygen committed:
clash: update to 1.18.0

Cyanoxygen committed:
syncthing: update to 1.24.0

Cyanoxygen committed:
frp: update to 0.51.3

Cyanoxygen committed:
xray: update to 1.8.4

Cyanoxygen committed:
dnscrypt: update to 2.1.5

Cyanoxygen committed:
v2ray: update to 5.7.0+git20230810

Cyanoxygen committed:
restic: update to 0.16.0

Cyanoxygen committed:
hugo: update to 0.118.2

Cyanoxygen committed:
direnv: update to 2.32.3

Cyanoxygen committed:
pcstat: update to 0.0.2

Cyanoxygen committed:
fzf: update to 0.42.0

Cyanoxygen committed:
hub: update to 2.14.2+git20230716

Cyanoxygen committed:
rclone: update to 1.63.1

Cyanoxygen committed:
git-lfs: disable mod vendoring

- Add asciidoctor as build dependency
- Adjust manual installation to adapt upstream manual directory change
- Fill build outputs

Cyanoxygen committed:
frp: add patch to revert upstream systemd unit removal

Cyanoxygen committed:
dnscrypt: update patches

Cyanoxygen committed:
poetry-code: update to 1.7.0

Cyanoxygen committed:
poetry-core: build using venv

- Clean up unused script.
- With -n switch on, PEP 517 build system will raise a BackendInvalid().
  However, after updating, this switch can be enabled.

Cyanoxygen committed:
poetry: update to 1.6.1

Cyanoxygen committed:
poetry: add missing dependency platformdirs

Cyanoxygen committed:
setuptools-scm: update to 7.1.0

Cyanoxygen committed:
setuptools-scm: disable Python 2

Cyanoxygen committed:
poetry-core: drop Python 2 support

- Since upstream moved to PEP 517.
- Add missing dependencies.

Cyanoxygen committed:
hatch-vcs: new, 0.3.0

Cyanoxygen committed:
scikit-build: new, 0.17.6

Cyanoxygen committed:
cython: update to 3.0.0

Cyanoxygen committed:
rapidfuzz-cpp: new, 2.0.0

Cyanoxygen committed:
rapidfuzz: new, 3.2.0

Cyanoxygen committed:
cleo: update to 2.0.1

Cyanoxygen committed:
cleo: add tomli, rapidfuzz as dependency

Cyanoxygen committed:
dulwich: new, 0.21.6

Cyanoxygen committed:
smmap: new, 5.0.0

Cyanoxygen committed:
poetry: add tomli, dulwich as dependnecy

Cyanoxygen committed:
gitdb: new, 4.0.10

Cyanoxygen committed:
gitpython: new, 3.1.35

Cyanoxygen committed:
mongo-tooling-metrics: new, 1.0.8

Cyanoxygen committed:
mongodb: update to 7.0.1

Cyanoxygen committed:
mongodb: add mongo-tooling-metrics as build depedency

Cyanoxygen committed:
mongodb: use gold instead of lld

- Or, fallback to bfd if gold is not available for the architecture.

Cyanoxygen committed:
mongodb: fix FTBFS on GCC 13

Cyanoxygen committed:
docker-compose: new, 2.21.0

Cyanoxygen committed:
groups/go-rebuilds: update

Cyanoxygen committed:
glide: bump REL for go 1.21.0 rebuild

Cyanoxygen committed:
dep: bump REL for go 1.21.0 rebuild

Cyanoxygen committed:
go-md2man: bump REL for go 1.21.0 rebuild

Cyanoxygen committed:
cf-tool: bump REL for go 1.21.0 rebuild

MingcongBai commented on app-containers/docker-compose/autobuild/build:
```diff
@@ -0,0 +1,13 @@
+	go build ${GOFLAGS} -trimpath -tags "${O_BUILDTAGS}" -ldflags "${GO_LDFLAGS}" -o "$SRCDIR"/bin/build/docker-compose ./cmd
+
+abinfo "Installing $PKGNAME ..."
+install -Dm755 "$SRCDIR"/bin/build/docker-compose "$PKGDIR"/usr/lib/docker/cli-plugins/docker-compose
```
Missing verbosity.

MingcongBai commented on app-database/mongodb/autobuild/build:
```diff
@@ -10,6 +10,12 @@ S_CXXFLAGS="$(echo "$CXXFLAGS" | tr -s ' ')"
 
+linker=gold
+if ab_match_arch riscv64 || ab_match_arch loongarch64 ; then
+	abinfo "Using bfd as linker instead of gold..."
```
Add a FIXME comment for missing gold support.

MingcongBai commented on app-devel/git-lfs/autobuild/build:
```diff
@@ -1,17 +1,24 @@
-	.
+go build -trimpath .
+
+
```
Extraneous empty line.

MingcongBai commented on app-devel/mongo-tooling-metrics/autobuild/defines:
```diff
@@ -0,0 +1,8 @@
+PKGNAME="mongo-tooling-metrics"
+PKGDES="A slim library which leverages Pydantic to reliably collect type enforced metrics and store them to MongoDB"
```
A library to collect type enforced metrics and store them in MongoDB databases

MingcongBai commented on runtime-vcs/dulwich/autobuild/defines:
```diff
@@ -0,0 +1,9 @@
+PKGNAME=dulwich
+PKGDES="Pure-Python Git implementation"
```
A Git implementation in Python

MingcongBai commented on runtime-common/rapidfuzz-cpp/spec:
```diff
@@ -0,0 +1,4 @@
+VER=2.0.0
+SRCS="https://github.com/maxbachmann/rapidfuzz-cpp/archive/refs/tags/v${VER}.tar.gz"
+CHKSUMS="sha256::0d6d399be1de151631bbc189b72089600884831a4dac91e22f17351cef18ae64"
+
```
Missing CHKUPDATE.

MingcongBai commented on runtime-vcs/gitpython/autobuild/defines:
```diff
@@ -0,0 +1,8 @@
+
+NOPYTHON2=1
+ABSPLITDBG=0
+
```
Extraneous blank line.

MingcongBai commented on runtime-vcs/gitpython/spec:
```diff
@@ -0,0 +1,4 @@
+VER=3.1.35
+SRCS="tbl::https://github.com/gitpython-developers/GitPython/archive/refs/tags/${VER}.tar.gz"
+CHKSUMS="sha256::cf9da1215b6bceee451a4dda8ddbaa90aa5fac42955c850f0679ba54d9762c26"
+
```
Missing CHKUPDATE.

MingcongBai commented on runtime-vcs/gitdb/spec:
```diff
@@ -0,0 +1,3 @@
+VER=4.0.10
+SRCS="tbl::https://github.com/gitpython-developers/gitdb/archive/refs/tags/${VER}.tar.gz"
+CHKSUMS="sha256::38b010cfae4b75576a7f99c60c1deb8a4c97c0477a6f07171e8e25a3a53c41ad"
```
Missing CHKUPDATE.

MingcongBai commented on runtime-vcs/gitdb/autobuild/defines:
```diff
@@ -0,0 +1,8 @@
+PKGNAME=gitdb
+PKGSEC=python
+PKGDES="IO of git-style object databases"
```
What does that even mean?

MingcongBai replied:
A Git object database for Python.

MingcongBai commented on runtime-vcs/smmap/spec:
```diff
@@ -0,0 +1,3 @@
+VER=5.0.0
+SRCS="tbl::https://github.com/gitpython-developers/smmap/archive/refs/tags/v${VER}.tar.gz"
+CHKSUMS="sha256::265a54c1cc32742ac37204396f62680bf45295fe03a4ece82f9e63a5a08c9686"
```
Missing CHKUPDATE.

MingcongBai commented on lang-python/rapidfuzz/autobuild/prepare:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Why the shebang?

MingcongBai commented on lang-python/rapidfuzz/autobuild/prepare:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+abinfo "Installing TaskFlow into project dependency..."
+rmdir "$SRCDIR/extern/taskflow"
```
Verbosity here.

MingcongBai commented on lang-python/poetry-core/autobuild/build:
```diff
@@ -0,0 +1,4 @@
+abinfo "Building poetry-core..."
+cd "$SRCDIR"
```
Why? Just use SRCDIR for the next two lines.

Cyanoxygen committed:
go: install binaries with prefix to avoid binary name collision

Cyanoxygen committed:
docker: update to 24.0.6+tini0.19.0

Cyanoxygen committed:
geoipupdate: rebase patch

Cyanoxygen committed:
conmon: update to 2.1.8

Cyanoxygen committed:
android-platform-tools: introduce hacks for ppc64el

- Vendored BoringSSL does not know about POWER, so we let it
  know.
- Android specific tools in vendored e2fsprogs are causing
  type conflicts, so we hack it for now.

Cyanoxygen committed:
criu: update to 3.18

Cyanoxygen committed:
criu: introduce patch to add missing defines

Cyanoxygen committed:
poetry: add pep517 related packages as build dependencies

Cyanoxygen committed:
conmon: add go as build dependency

Cyanoxygen committed:
criu: define __SANE_USERSPACE_TYPES__ for loongson3
